keras/src/layers/preprocessing/stft_spectrogram.py

```diff
@@ -0,0 +1,410 @@
+            `frame_length`. Defaults to the smallest power of two that is
+            greater than or equal to `frame_length`.
+        window: (String or array_like), the windowing function to apply to each
+            frame. Can be `"hann`" (default), `"hamming`", or a custom window
+            provided as an array_like.
+        periodic: Boolean, if True, the window function will be treated as
+            periodic. Defaults to `False`.
+        scaling: String, type of scaling applied to the window. Can be
+            `"density`", `"spectrum`", or None. Default is `"density`".
+        padding: String, padding strategy. Can be `"valid`" or `"same`".
+            Defaults to `"valid"`.
+        expand_dims: Boolean, if True, will expand the output into spectrograms
+            into two dimensions to be compatible with image models.
+            Defaults to `False`.
+        data_format: String, either `"channels_last"` or `"channels_first"`.
+            The ordering of the dimensions in the inputs. `"channels_last"`
+            corresponds to inputs with shape `(batch, height, width, channels)`
+            while `"channels_first"` corresponds to inputs with shape
+            `(batch, channels, height, weight)`. Defaults to `"channels_last"`.
+
+    Raises:
+        ValueError: If an invalid value is provided for `"mode`", `"scaling`",
+            `"padding`", or other input arguments.
+        TypeError: If the input data type is not one of `"float16`",
+            `"float32`", or `"float64`".
+
+    Input shape:
+        A 3D tensor of shape `(batch_size, time_length, input_channels)`, if
+        `data_format=="channels_last"`, and of shape
+        `(batch_size, input_channels, time_length)` if
+        `data_format=="channels_first"`, where `time_length` is the length of
+        the input signal, and `input_channels` is the number of input channels.
+        The same kernels are applied to each channel independetly.
+
+    Output shape:
+        If `data_format=="channels_first" and not expand_dims`, a 3D tensor:
+            `(batch_size, input_channels * freq_channels, new_time_length)`
+        If `data_format=="channels_last" and not expand_dims`, a 3D tensor:
+            `(batch_size, new_time_length, input_channels * freq_channels)`
+        If `data_format=="channels_first" and expand_dims`, a 4D tensor:
+            `(batch_size, input_channels, new_time_length, freq_channels)`
+        If `data_format=="channels_last" and expand_dims`, a 4D tensor:
+            `(batch_size, new_time_length, freq_channels, input_channels)`
+
+        where `new_time_length` depends on the padding, and `freq_channels` is
+        the number of FFT bins `(fft_length // 2 + 1)`.
+
+    Example:
+        ```python
+        spectrogram_layer = keras.layers.STFTSpectrogram(
+            mode='log', frame_length=256, fft_length=512
+        )
+        output = spectrogram_layer(input_signal)
+        ```
+
```

Comment:
Since this layer is fairly niche, please illustrate its usage with detailed code examples (with descriptions of what they do and what they'd be used for), e.g. "different modes of output", "non trainable init vs fine-tuning", etc.

The user should be able to read the docstring and understand: what is this layer for? when would I need it? how do I use it?

Reply:
I added now three code examples demonstrating the different use cases. Also check the more comprehensive tutorial I added [here](https://github.com/keras-team/keras-io/pull/1947).